Upgrade zlib

The zlib version used during building is not compatible with Apple Clang 17.0.0 on macOS Tahoe. I see errors like:
```
  ERROR: /private/var/folders/jv/cgkfvslj6nq1l7cw0c8c_8gm0000gn/T/spack9l9gotwq/_bazel_Adam/e634232a13c636b9e832565593c25887/external/zlib/BUILD.bazel:37:11: Compiling zutil.c [for tool] failed: (Exit 1): cc_wrapper.sh failed: error executing CppCompile command (from target @@zlib//:zlib) external/bazel_tools~cc_configure_extension~local_config_cc/cc_wrapper.sh -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object ... (remaining 34 arguments skipped)

  Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
  In file included from external/zlib/zutil.c:8:
  external/zlib/zutil.h:170:11: warning: 'OS_CODE' macro redefined [-Wmacro-redefined]
    170 | #  define OS_CODE 19
        |           ^
  external/zlib/zutil.h:141:11: note: previous definition is here
    141 | #  define OS_CODE  7
        |           ^
  In file included from external/zlib/zutil.c:10:
  In file included from external/zlib/gzguts.h:21:
  In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdio.h:61:
  /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_stdio.h:322:7: error: expected identifier or '('
    322 | FILE    *fdopen(int, const char *) __DARWIN_ALIAS_STARTING(__MAC_10_6, __IPHONE_2_0, __DARWIN_ALIAS(fdopen));
        |          ^
  external/zlib/zutil.h:147:33: note: expanded from macro 'fdopen'
    147 | #        define fdopen(fd,mode) NULL /* No fdopen() */
        |                                 ^
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/17/include/__stddef_null.h:26:16: note: expanded from macro 'NULL'
     26 | #define NULL ((void*)0)
        |                ^
  In file included from external/zlib/zutil.c:10:
  In file included from external/zlib/gzguts.h:21:
  In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdio.h:61:
  /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_stdio.h:322:7: error: expected ')'
  external/zlib/zutil.h:147:33: note: expanded from macro 'fdopen'
    147 | #        define fdopen(fd,mode) NULL /* No fdopen() */
        |                                 ^
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/17/include/__stddef_null.h:26:16: note: expanded from macro 'NULL'
     26 | #define NULL ((void*)0)
        |                ^
  /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_stdio.h:322:7: note: to match this '('
```

--- a/.bazelrc
+++ b/.bazelrc
@@ -1,0 +1,22 @@
+# Bazel configuration for C++ standard and build options
+
+# Enforce C++17 standard for both target and host (tools) compilation
+build --cxxopt=-std=gnu++17
+build --host_cxxopt=-std=gnu++17
+build --cxxopt=-fno-strict-aliasing
+build --host_cxxopt=-fno-strict-aliasing
+
+# Disable bzlmod (new module system) to use traditional WORKSPACE
+common --noenable_bzlmod
+
+# Don't keep state after build to prevent cache issues
+common --nokeep_state_after_build
+
+# Verbose logging for CI debugging
+build --verbose_failures
+
+# macOS specific settings
+build:macos --cxxopt=-std=gnu++17
+build:macos --host_cxxopt=-std=gnu++17
+build:macos --cxxopt=-fno-strict-aliasing
+build:macos --host_cxxopt=-fno-strict-aliasing

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -30,6 +30,18 @@ jobs:
         cache-dependency-path: |
           setup.py
 
+    - name: Install Bazel
+      run: |
+        mkdir -p ~/.local/bin
+        if [[ "${{ runner.os }}" == "Linux" ]]; then
+          wget --no-check-certificate -q https://github.com/bazelbuild/bazel/releases/download/${{ env.USE_BAZEL_VERSION }}/bazel-${{ env.USE_BAZEL_VERSION }}-linux-x86_64 -O ~/.local/bin/bazel
+        elif [[ "${{ runner.os }}" == "macOS" ]]; then
+          wget --no-check-certificate -q https://github.com/bazelbuild/bazel/releases/download/${{ env.USE_BAZEL_VERSION }}/bazel-${{ env.USE_BAZEL_VERSION }}-darwin-arm64 -O ~/.local/bin/bazel
+        fi
+        chmod +x ~/.local/bin/bazel
+        echo "$HOME/.local/bin" >> $GITHUB_PATH
+        bazel --version
+
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -13,12 +13,11 @@ env:
 
 jobs:
   tests:
-    runs-on: ubuntu-latest
-
-
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: ['3.9', '3.10', '3.11', '3.12']
+        os: [ubuntu-latest, macos-latest]
+        python-version: ['3.9', '3.10', '3.11', '3.12', '3.13']
 
     steps:
     - uses: actions/checkout@v4

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -9,7 +9,7 @@ on:
   workflow_dispatch:
 
 env:
-  USE_BAZEL_VERSION: "7.6.1"
+  USE_BAZEL_VERSION: "6.5.0"
 
 jobs:
   tests:

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -7,7 +7,7 @@ on:
     types: [published]
 
 env:
-  USE_BAZEL_VERSION: "7.6.1"
+  USE_BAZEL_VERSION: "6.5.0"
 
 
 jobs:
@@ -35,6 +35,18 @@ jobs:
     - name: Install python build dependencies
       run: |
           python -m pip install --upgrade pip "setuptools<70" "build<2.0" wheel
+
+    - name: Install Bazel
+      run: |
+        mkdir -p ~/.local/bin
+        if [[ "${{ runner.os }}" == "Linux" ]]; then
+          wget --no-check-certificate -q https://github.com/bazelbuild/bazel/releases/download/${{ env.USE_BAZEL_VERSION }}/bazel-${{ env.USE_BAZEL_VERSION }}-linux-x86_64 -O ~/.local/bin/bazel
+        elif [[ "${{ runner.os }}" == "macOS" ]]; then
+          wget --no-check-certificate -q https://github.com/bazelbuild/bazel/releases/download/${{ env.USE_BAZEL_VERSION }}/bazel-${{ env.USE_BAZEL_VERSION }}-darwin-arm64 -O ~/.local/bin/bazel
+        fi
+        chmod +x ~/.local/bin/bazel
+        echo "$HOME/.local/bin" >> $GITHUB_PATH
+        bazel --version
 
     - name: Build wheels
       run: |
@@ -79,7 +91,6 @@ jobs:
         uses: pypa/gh-action-pypi-publish@release/v1.12
         with:
           packages-dir: wheels/
-          repository-url: https://pypi.org/legacy/
           # already checked, and the pkginfo/twine versions on this runner causes check to fail
           verify-metadata: false
           verbose: true

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -18,8 +18,8 @@ jobs:
     strategy:
       # fail-fast: true
       matrix:
-        os: [ubuntu]
-        python-version: ['3.9', '3.10', '3.11', '3.12']
+        os: [ubuntu, macos]
+        python-version: ['3.9', '3.10', '3.11', '3.12', '3.13']
 
 
     runs-on: ${{ format('{0}-latest', matrix.os) }}

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -34,7 +34,7 @@ jobs:
 
     - name: Install python build dependencies
       run: |
-          python -m pip install --upgrade pip build
+          python -m pip install --upgrade pip "setuptools<70" "build<2.0" wheel
 
     - name: Build wheels
       run: |

--- a/.gitignore
+++ b/.gitignore
@@ -2,7 +2,6 @@ bazel-bin
 bazel-metadata
 bazel-out
 bazel-testlogs
-tensorflow_metadata.egg-info/
 tensorflow_metadata/proto/v0/anomalies_pb2.py
 tensorflow_metadata/proto/v0/derived_feature_pb2.py
 tensorflow_metadata/proto/v0/metric_pb2.py
@@ -10,3 +9,6 @@ tensorflow_metadata/proto/v0/path_pb2.py
 tensorflow_metadata/proto/v0/problem_statement_pb2.py
 tensorflow_metadata/proto/v0/schema_pb2.py
 tensorflow_metadata/proto/v0/statistics_pb2.py
+build/
+dist/
+*.egg-info/

--- a/.gitignore
+++ b/.gitignore
@@ -1,0 +1,12 @@
+bazel-bin
+bazel-metadata
+bazel-out
+bazel-testlogs
+tensorflow_metadata.egg-info/
+tensorflow_metadata/proto/v0/anomalies_pb2.py
+tensorflow_metadata/proto/v0/derived_feature_pb2.py
+tensorflow_metadata/proto/v0/metric_pb2.py
+tensorflow_metadata/proto/v0/path_pb2.py
+tensorflow_metadata/proto/v0/problem_statement_pb2.py
+tensorflow_metadata/proto/v0/schema_pb2.py
+tensorflow_metadata/proto/v0/statistics_pb2.py

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -26,9 +26,9 @@ http_archive(
 http_archive(
     name = "zlib",
     build_file = "@com_google_protobuf//:third_party/zlib.BUILD",
-    sha256 = "d8688496ea40fb61787500e863cc63c9afcbc524468cedeb478068924eb54932",
-    strip_prefix = "zlib-1.2.12",
-    urls = ["https://github.com/madler/zlib/archive/v1.2.12.tar.gz"],
+    sha256 = "17e88863f3600672ab49182f217281b6fc4d3c762bde361935e436a95214d05c",
+    strip_prefix = "zlib-1.3.1",
+    urls = ["https://github.com/madler/zlib/archive/v1.3.1.tar.gz"],
 )
 
 # Needed by com_google_protobuf.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,0 +1,3 @@
+[build-system]
+requires = ["setuptools<70", "wheel"]
+build-backend = "setuptools.build_meta"

--- a/setup.py
+++ b/setup.py
@@ -124,6 +124,8 @@ setup(
         "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
         "Programming Language :: Python :: 3.11",
+        "Programming Language :: Python :: 3.12",
+        "Programming Language :: Python :: 3.13",
         "Programming Language :: Python :: 3 :: Only",
         "Topic :: Scientific/Engineering :: Artificial Intelligence",
         "Topic :: Scientific/Engineering :: Mathematics",

